test/unit/storage/index.spec.ts

```diff
@@ -69,5 +77,56 @@ describe('Storage', () => {
       const storage2: Storage = getStorage(mockApp);
       expect(storage1).to.equal(storage2);
     });
+    it('should return an error when no metadata is available', async () => {
+      sandbox
+        .stub(StorageUtils, 'getFirebaseMetadata')
+        .returns(Promise.resolve({} as StorageUtils.FirebaseMetadata));
+      const storage1 = getStorage(mockApp);
+      const fileRef = storage1.bucket('gs://mock').file('abc');
+      await expect(getDownloadUrl(fileRef)).to.be.rejectedWith(
+        'No download token available. Please create one in the Firebase Console.'
+      );
+    });
+    it('should return an error when unable to fetch metadata', async () => {
+      const error = new Error('Something went wrong');
+      sandbox
+        .stub(StorageUtils, 'getFirebaseMetadata')
+        .returns(Promise.reject(error));
+      const storage1 = getStorage(mockApp);
+      const fileRef = storage1.bucket('gs://mock').file('abc');
+      await expect(getDownloadUrl(fileRef)).to.be.rejectedWith(
+        error
+      );
+    });
+    it('should return the proper download url when metadata is available', async () => {
+      const downloadTokens = ['abc', 'def'];
+      sandbox
+        .stub(StorageUtils, 'getFirebaseMetadata')
+        .returns(
+          Promise.resolve({
+            downloadTokens: downloadTokens.join(','),
+          } as StorageUtils.FirebaseMetadata)
+        );
+      const storage1 = getStorage(mockApp);
+      const fileRef = storage1.bucket('gs://mock').file('abc');
+      await expect(getDownloadUrl(fileRef)).to.eventually.eq(
+        `https://firebasestorage.googleapis.com/v0/b/${fileRef.bucket.name}/o/${encodeURIComponent(fileRef.name)}?alt=media&token=${downloadTokens[0]}`
+      );
+    });
+    it('should use the emulator host name when process.env is set', async () => {
+      const HOST = 'https://localhost:9091';
+      process.env.STORAGE_EMULATOR_HOST = HOST;
```

Comment:
We use `FIREBASE_STORAGE_EMULATOR_HOST` as the environment variable for Firebase and map it to `STORAGE_EMULATOR_HOST` internally in the Admin SDK. Please test for `FIREBASE_STORAGE_EMULATOR_HOST` as well.